tests/net/mgmt/src/mgmt.c

```diff
@@ -423,4 +423,32 @@ ZTEST(mgmt_fn_test_suite, test_mgmt)
 		      "test_synchronous_event_listener failed");
 }
 
+K_SEM_DEFINE(wait_for_event_processing, 0, 1);
```

Comment:
Please use `static K_SEM_DEFINED(...`
Also we could check that the callback is only called once, so I would add here
```
static int cb_call_count;
```